lib/index.ts

```diff
@@ -27,26 +44,161 @@ export function inspect(root, targetFile, options) {
     baseargs = ['run', 'python'];
   }
 
-  return Promise.all([
+  const [plugin, pkg] = await Promise.all([
     getMetaData(command, baseargs, root, targetFile),
     getDependencies(
       command,
       baseargs,
       root,
       targetFile,
-      options.allowMissing,
+      options.allowMissing || false,
       includeDevDeps,
       options.args
     ),
-  ]).then((result) => {
-    return {
-      plugin: result[0],
-      package: result[1],
-    };
+  ]);
+  return { plugin, package: pkg };
+}
+
+interface UpgradeRemediation {
+  upgradeTo: string;
+  // Other fields are of no interest
+}
+
+interface DependencyUpdates {
+  [from: string]: UpgradeRemediation;
+}
+
+interface ManifestFiles {
+  // Typically these are requirements.txt, constraints.txt and Pipfile
+  [name: string]: string; // content
```

Comment:
Is it worth making this strict? We know what we support, so making it an arbitrary key seems unnecessary.